tools/link.py

```diff
@@ -163,6 +163,10 @@ def base64_encode(filename):
   return b64.decode('ascii')
 
 
+def base64_or_binary_encode(b):
+  return binary_encode(b) if settings.SINGLE_FILE and settings.SINGLE_FILE_BINARY_ENCODE else base64_encode(b)
```

Comment:
I think you can remove `settings.SINGLE_FILE` here (make it an assert if you like)